src/cc_catalog_airflow/dags/provider_api_scripts/test_smithsonian.py

```diff
@@ -0,0 +1,1005 @@
+import json
+import logging
+import os
+from unittest.mock import patch, call
+
+import pytest
+
+import smithsonian as si
+
+logger = logging.getLogger(__name__)
+
+logging.basicConfig(
+    format='%(asctime)s - %(name)s - %(levelname)s:  %(message)s',
+    level=logging.DEBUG
+)
+
+RESOURCES = os.path.join(
+    os.path.abspath(os.path.dirname(__file__)), 'tests/resources/smithsonian'
+)
+
+
+def test_get_hash_prefixes_with_len_one():
+    expect_prefix_list = [
+        '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd',
+        'e', 'f'
+    ]
+    actual_prefix_list = list(si._get_hash_prefixes(1))
+    assert actual_prefix_list == expect_prefix_list
+
+
+@pytest.mark.parametrize(
+    'input_int,expect_len,expect_first,expect_last',
+    [
+        (1, 16, '0', 'f'),
+        (2, 256, '00', 'ff'),
+        (3, 4096, '000', 'fff'),
+        (4, 65536, '0000', 'ffff'),
+    ],
+)
+def test_get_hash_prefixes_with_other_len(
+        input_int,
+        expect_len,
+        expect_first,
+        expect_last
+):
+    actual_list = list(si._get_hash_prefixes(input_int))
+    assert all('0x' not in h for h in actual_list)
+    assert all(
+        int(actual_list[i + 1], 16) - int(actual_list[i], 16) == 1
+        for i in range(len(actual_list) - 1)
+    )
+    assert len(actual_list) == expect_len
+    assert actual_list[0] == expect_first
+    assert actual_list[-1] == expect_last
+
+
+def test_process_hash_prefix_with_none_response_json():
+    endpoint = 'https://abc.com/123'
+    limit = 100
+    hash_prefix = '00'
+    retries = 3
+    qp = {'q': 'abc'}
+
+    patch_get_response_json = patch.object(
+        si.delayed_requester,
+        'get_response_json',
+        return_value=None
+    )
+    patch_build_qp = patch.object(
+        si,
+        '_build_query_params',
+        return_value=qp
+    )
+    patch_process_response = patch.object(
+        si,
+        '_process_response_json'
+    )
+    with\
+            patch_get_response_json as mock_get_response_json,\
+            patch_build_qp as mock_build_qp,\
+            patch_process_response as mock_process_response:
```

Comment:
I completely agree.  Unfortunately, python `with` statements don't allow the same parentheses use as (for example) `if` statements.  See

https://bugs.python.org/issue12782

if you're curious about it.

The best alternative I'm aware of to these backslashes is:
```python
    with (
            patch_get_response_json) as mock_get_response_json, (
            patch_build_qp) as mock_build_qp, (
            patch_process_response) as mock_process_response:
```
I find this harder, not easier to read since it kind of splits the objects from their variables (the `as` in a confusing way.

Reply:
I suppose another option is
```python
    with (
            patch_get_response_json
    ) as mock_get_response_json, (
            patch_build_qp
    ) as mock_build_qp, (
            patch_process_response
    ) as mock_process_response:
```
but I think that's even more confusing.  The indentation is almost logically backwards at that point.